### ISSUE TITLE
MYSQL

### ISSUE THREAD

**Issue by lowoa** [SmItH197/SteamAuthentication#258]
How to make login page with insert to mysql database?

**Comment by LeventHAN** [comment]
> How to make login page with insert to mysql database?

Hey mate this is not an issue, so please do not open issue tickets with question on how to make things, this is not StackOverflow. But just to answer your question since it is easy.

Once the user is logged in and is redirected to your website you can start to do what you want.

In short that would be (please figure yourself out on how to handle users that already exist on the DB... It wouldnt be that hard 👍 ):
```php
            if (!isset($_SESSION['steamid'])) {
              // User is not logged in. So we show the login button.
              echo '<div class="mt-3">';
              echo "<a href='?login'><img class='mr-4' src='https://steamcommunity-a.akamaihd.net/public/images/signinthroughsteam/sits_02.png' width='78'></a>";
              echo '</div>';
           } else {
             // User is logged in. So we start our INSERT.
             include('steamauth/userInfo.php');
             $conn = mysqli_connect("YOUR_DB_IP", "YOUR_DB_USERNAME", "YOUR_DB_PASS", "YOUR_DB_NAME") or die("Error " . mysqli_error($conn);
             // Always use prepared statements 👍 + this DB structure is just something I made up, you can fill your data here.
             $stmt = $conn->prepare("INSERT INTO steamPlayers(id, steamid, communityvisibilitystate, profilestate, personaname, lastlogoff, profileurl, personastate, realname, primaryclanid, timecreated, uptodate, avatar, avatarmedium, avatarfull) VALUES (DEFAULT, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)");
             // i binds to ?
             $stmt->bind_param("iiiiiiiiiiiiii", $steamprofile['steamid'], $steamprofile['communityvisibilitystate'], $steamprofile['profilestate'], $steamprofile['personaname'], $steamprofile['lastlogoff'], $steamprofile['profileurl'], $steamprofile['personastate'], $steamprofile['realname'], $steamprofile['primaryclanid'], $steamprofile['timecreated'], $steamprofile['uptodate'], $steamprofile['avatar'], $steamprofile['avatarmedium'], $steamprofile['avatarfull'], );
             if (!$stmt->execute()) {
                echo $stmt->error;
             } else {
                echo 'Done.';
             }
             $conn->close();
           }
```

If you don't know any coding you probably will not understand the above... 
However I do need to say the above code is not a copy-paste code!

**Comment by BlackCetha** [comment]
This library is not designed to handle your database work. Looking past the profanity in the above post, there are still some points factually wrong. For one, it will insert a new entry into your database everytime a user (even a known one) logs in. It lacks proper error handling and will spill out your code on errors. Not too sure about this last point, but it may even expose your database credentials under certain conditions.

**Please read up on how to properly save user data separately from this library and do not use the above code.**

**Comment by LeventHAN** [comment]
> This library is not designed to handle your database work. Looking past the profanity in the above post, there are still some points factually wrong. For one, it will insert a new entry into your database everytime a user (even a known one) logs in. It lacks proper error handling and will spill out your code on errors. Not too sure about this last point, but it may even expose your database credentials under certain conditions.
> 
> **Please read up on how to properly save user data separately from this library and do not use the above code.**

Indeed! I did try to underline this with my last message: `However I do need to say the above code is not a copy-paste code!`
and also: `please figure yourself out on how to handle users that already exist` (For the mysql_connect I didn't use vars since it would make the example longer but indeed it is not secure it would show the credintials on a error)
I don't know how they call those codes but it is a example and not ready for production.
But thanks for making this clear since I think I didn't make it clear enough  👍
